File: apzkr-pzpi-21-11-milennyi-ivan/Task3-WebClient/src/viewModels/FeedVM.ts
// Клас для відображення інформації про корм
export class FeedVM {
    id: number; // Ідентифікатор корму
    amount: number; // Кількість корму в кг
    name: string; // Назва корму
    calories: number; // Кількість калорій
    fat: number; // Кількість жиру
    protein: number; // Кількість білків
    carbohydrates: number; // Кількість вуглеводів
    breedName: string | null; // Назва породи, для якої призначений корм
    sheepCount: number; // Кількість овець, які споживають цей корм

    constructor(id: number, amount: number, name: string, calories: number, fat: number, protein: number, carbohydrates: number, breedName: string | null, sheepCount: number) {
        this.id = id;
        this.amount = amount / 1000;
        this.name = name;
        this.calories = calories;
        this.fat = fat;
        this.protein = protein;
        this.carbohydrates = carbohydrates;
        this.breedName = breedName;
        this.sheepCount = sheepCount;
    }
}

File: apzkr-pzpi-21-11-milennyi-ivan/Task1-Server/src/view_models/feed_vm.rs
use serde::Serialize;
use sqlx::FromRow;
use utoipa::ToSchema;

// Структура, що представляє представлення корму (ViewModel)
#[derive(Debug, Serialize, FromRow, ToSchema)]
pub(crate) struct FeedVM {
    id: u64, // Ідентифікатор корму
    amount: u32, // Кількість корму
    name: String, // Назва корму
    calories: u32, // Калорійність корму
    fat: u32, // Вміст жиру
    protein: u32, // Вміст білків
    carbohydrates: u32, // Вміст вуглеводів
    breed_name: Option<String>, // Назва породи, для якої призначений корм
    sheep_count: Option<u64>, // Кількість овець, що споживають цей корм
}